xbmc/view/GUIViewState.cpp

```diff
@@ -344,6 +345,37 @@ void CGUIViewState::SetSortMethod(SortDescription sortDescription)
   return SetSortMethod(sortDescription.sortBy, sortDescription.sortOrder);
 }
 
+SortDescription CGUIViewState::ChooseSortMethod()
+{
+  
+  CGUIDialogSelect *dialog = (CGUIDialogSelect *)g_windowManager.GetWindow(WINDOW_DIALOG_SELECT);
+  if (dialog)
+  {
+    dialog->SetHeading(CVariant{ 32104 }); // Label "Sort by"
+    int iIndex = 0;
+    int iSelected = 0;
+    for (int i = 0; i < m_sortMethods.size(); ++i)
+    {
+      dialog->Add(g_localizeStrings.Get(m_sortMethods[i].m_buttonLabel));
+      if (i == m_currentSortMethod)
+        iSelected = iIndex;
+      iIndex++;
+    }
```

Comment:
All the indices here seems unnecessary.. Simply pass m_currentSortMethod to SetSelected